# Poswork 4
En este paso, se comenzó el análisis del conjunto de datos que fue seleccionado en la sección anterior. 
Para este primer análisis, se utiliza la biblioteca Pandas.

El conjunto de datos representa el listado de propiedades Airbnb de las ciudades... (describir)



Para hacer uso de Pandas y Numpy es nesario importar las librerías.

Para mayor información sobre [Pandas](https://pandas.pydata.org/docs/) y [Numpy](https://numpy.org/doc/) consultar la documentación.


In [1]:
#Importando librerias
import pandas as pd
import numpy as np

El conjunto de datos se encuentra almacedo en Google Drive, es necesario habilitar el acceso a Google Colaboratory con la finalidad de cargar los datos. Una forma de realizar este paso es ejecutar la siguiente línea. 

In [2]:
from google.colab import drive
drive.mount('https://github.com/BeduDSEquipo9/G2DSF2_Python/blob/Developer/Datasets/airbnb.csv')

ModuleNotFoundError: No module named 'google'

El conjunto de datos esta en formato csv, por lo que se utiliza `pd.read_csv()` para crear un objeto de tipo DataFrame.

In [ ]:
data=pd.read_csv('/content/drive/MyDrive/BEDUDSF2M3Python2023Proy/Proy/Datasets/airbnb.csv')

Una vez cargados los datos en un DataFrame comenzaremos la exploración de los datos. 

Una primera vista a los datos se realiza con la ayuda de `head()`, que muestra los defecto los primeros diez regitros del DataFrame. Algo simiar con el método `tail()` que muesta los últimos registros del DataFrame.





In [ ]:
data.head()

id property_type        room_type  \
0  46459         House     Private room   
1  16468     Apartment  Entire home/apt   
2  30190     Apartment  Entire home/apt   
3  43181     Townhouse     Private room   
4  62109   Condominium  Entire home/apt   

                                           amenities  accommodates  bathrooms  \
0  {"Wireless Internet","Air conditioning",Kitche...             2        0.0   
1  {Internet,"Wireless Internet",Kitchen,Heating,...             2        1.0   
2  {TV,Internet,"Wireless Internet","Air conditio...             4        1.0   
3  {"Wireless Internet","Air conditioning","Pets ...             3        1.0   
4  {TV,"Cable TV",Internet,"Wireless Internet","A...             5        2.0   

   bed_type cancellation_policy  cleaning_fee     city  ...   longitude  \
0  Real Bed              strict          True      NYC  ...  -74.003574   
1  Real Bed              strict          True       LA  ... -118.474525   
2  Real Bed              strict          True      NYC  ...  -73.993318   
3  Real Bed              strict          True       LA  ... -118.096453   
4  Real Bed              strict          True  Chicago  ...  -87.606271   

                                                name neighbourhood  \
0                 AFFORDABLE & COZY IN SUNSET PARK T   Sunset Park   
1  Luxurious and quiet - 30 second walk to the be...        Venice   
2                         Bensonhurst STYLISH STUDIO   Bensonhurst   
3  LA San Gabriel  Suit room 圣盖博大套房 免费停车 Free Par...   San Gabriel   
4  XTRA Large 2bdrm 2bath in the City near "El" T...      Woodlawn   

  number_of_reviews review_scores_rating  \
0                15                 89.0   
1                12                 98.0   
2                26                 94.0   
3                33                 95.0   
4                60                 88.0   

                                       thumbnail_url zipcode bedrooms  beds  \
0  https://a0.muscache.com/im/pictures/105652616/...   11232      1.0   1.0   
1                                                NaN   90291      1.0   1.0   
2  https://a0.muscache.com/im/pictures/bf8ee448-b...   11214      0.0   2.0   
3  https://a0.muscache.com/im/pictures/0cff238c-a...   91776      1.0   2.0   
4  https://a0.muscache.com/im/pictures/92108903/0...   60637      2.0   2.0   

   price  
0   50.0  
1  199.0  
2   85.0  
3   70.0  
4   72.0  

[5 rows x 29 columns]

In [ ]:
data.tail()

id property_type        room_type  \
59283  64666         House  Entire home/apt   
59284  10864   Condominium  Entire home/apt   
59285  14978     Apartment  Entire home/apt   
59286  58971     Apartment  Entire home/apt   
59287  35007         House  Entire home/apt   

                                               amenities  accommodates  \
59283  {TV,"Cable TV",Internet,"Wireless Internet","A...             5   
59284  {TV,"Cable TV","Wireless Internet","Air condit...             2   
59285  {TV,Internet,"Wireless Internet","Air conditio...             3   
59286  {TV,"Wireless Internet","Air conditioning","Wh...             3   
59287  {TV,"Cable TV",Internet,"Wireless Internet","A...            14   

       bathrooms  bed_type cancellation_policy  cleaning_fee city  ...  \
59283        2.0  Real Bed            moderate          True   LA  ...   
59284        1.5  Real Bed              strict          True  NYC  ...   
59285        1.0  Real Bed            moderate          True   LA  ...   
59286        1.0  Real Bed              strict          True  NYC  ...   
59287        3.0  Real Bed              strict          True   LA  ...   

        longitude                                              name  \
59283 -118.501966                  Lake Balboa House / Zen Retreat!   
59284  -73.952439             Modern Beautiful Duplex in Brownstone   
59285 -118.444475                     Artsy and Spacious Back House   
59286  -73.939087               Cozy Apartment off J&M Subway Lines   
59287 -118.377014  KOSHER-FRIENDLY 4 BEDROOM SLEEPS 10 and up to 12   

          neighbourhood number_of_reviews review_scores_rating  \
59283       Lake Balboa                 5                 88.0   
59284            Harlem                11                100.0   
59285  West Los Angeles                 0                  NaN   
59286          Bushwick                11                100.0   
59287               NaN                 6                 93.0   

                                           thumbnail_url zipcode bedrooms  \
59283  https://a0.muscache.com/im/pictures/61891002/2...   91406      3.0   
59284  https://a0.muscache.com/im/pictures/8dda462d-6...   10027      1.0   
59285  https://a0.muscache.com/im/pictures/d807e712-d...   90064      1.0   
59286  https://a0.muscache.com/im/pictures/0246b6ff-f...   11206      1.0   
59287                                                NaN   90035      4.0   

       beds  price  
59283   3.0  225.0  
59284   1.0  200.0  
59285   2.0  135.0  
59286   1.0  115.0  
59287   5.0  360.0  

[5 rows x 29 columns]

El conjunto de datos Airbnb contiene 59,288 observaciones y 29 variables, de acuerdo don el resultado obtenido con el atributo `shape`

In [ ]:
data.shape

(59288, 29)

El atributo `columns` permite obtener el nombre de cada una de las columnas del conjunto de datos.

In [ ]:
data.columns

Index(['id', 'property_type', 'room_type', 'amenities', 'accommodates',
       'bathrooms', 'bed_type', 'cancellation_policy', 'cleaning_fee', 'city',
       'description', 'first_review', 'host_has_profile_pic',
       'host_identity_verified', 'host_response_rate', 'host_since',
       'instant_bookable', 'last_review', 'latitude', 'longitude', 'name',
       'neighbourhood', 'number_of_reviews', 'review_scores_rating',
       'thumbnail_url', 'zipcode', 'bedrooms', 'beds', 'price'],
      dtype='object')

En resultado de la línea anterior muestra que los nombres de las columnas utilizan de manera estándarizada un formato de escritura `snake_case`. Además, se puede apreciar que cada uno de los nombres es representativo del dato que almacenan.

Para completar la información se utiliza el métodol `info()` se puede cononcer más detalles del dataset.



In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59288 entries, 0 to 59287
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      59288 non-null  int64  
 1   property_type           59288 non-null  object 
 2   room_type               59288 non-null  object 
 3   amenities               59288 non-null  object 
 4   accommodates            59288 non-null  int64  
 5   bathrooms               59133 non-null  float64
 6   bed_type                59288 non-null  object 
 7   cancellation_policy     59288 non-null  object 
 8   cleaning_fee            59288 non-null  bool   
 9   city                    59288 non-null  object 
 10  description             59288 non-null  object 
 11  first_review            46626 non-null  object 
 12  host_has_profile_pic    59137 non-null  object 
 13  host_identity_verified  59137 non-null  object 
 14  host_response_rate      44730 non-null

A continuación se utiliza el método `unique()` para conocer los valores unicos de algunas de las columnas que contienen tipos de datos categóricos:

In [ ]:
data['property_type'].unique()

array(['House', 'Apartment', 'Townhouse', 'Condominium', 'Villa', 'Loft',
       'Bed & Breakfast', 'Guesthouse', 'Other', 'Boutique hotel',
       'Bungalow', 'Cabin', 'Boat', 'Hostel', 'Camper/RV', 'Guest suite',
       'Dorm', 'Vacation home', 'In-law', 'Tent', 'Treehouse',
       'Timeshare', 'Castle', 'Yurt', 'Tipi', 'Earth House',
       'Serviced apartment', 'Hut', 'Island', 'Chalet', 'Train', 'Cave',
       'Lighthouse'], dtype=object)

In [ ]:
data['room_type'].unique()

array(['Private room', 'Entire home/apt', 'Shared room'], dtype=object)

In [ ]:
data['bed_type'].unique()

array(['Real Bed', 'Futon', 'Airbed', 'Pull-out Sofa', 'Couch'],
      dtype=object)

In [ ]:
data['cancellation_policy'].unique()

array(['strict', 'flexible', 'moderate', 'super_strict_30',
       'super_strict_60'], dtype=object)

Despues de ejecutar las líneas anterioers, se resumen que las 29 variable son las siguientes:

* `id` que es un valor numérico que identifica la propiedad lista en Airbnb y es de tipo `int64` 
* `property_type` es una valor categórico nominal que indica el tipo de propiedad: `House`, `Apartment`, `Townhouse`, `Condominium`, `Villa`, `Loft`, `Bed & Breakfast`, `Guesthouse`, `Other`, `Boutique hotel`,  `Bungalow`, `Cabin`, `Boat`, `Hostel`, `Camper/RV`, `Guest suite`,  `Dorm`, `Vacation home`, `In-law`, `Tent`, `Treehouse`,  `Timeshare`, `Castle`, `Yurt`, `Tipi`, `Earth House`,  `Serviced apartment`, `Hut`, `Island`, `Chalet`, `Train`, `Cave`,  `Lighthouse`. Es de tipo cadena de caracteres.
*  `room_type` es un valor categórico nominal que indica el tipo de habitación. Por ejemplo, pude tomar valores como: `Entire home/apt`, `Private room` y `Shared room`.
* `amenities`, las amenidades  son todos aquellos espacios o instalaciones dentro de una propiedad capaces de proporcionar una mejor calidad de vida. Por ende, son características que proporcionan un mayor valor al inmueble. El dato esta almacena en formato de cadena, donde se separan por coma cada una de las amenidades. Ejemplo: `{"TV", "Wireless Internet", "Air conditioning", "Wheelchair accessible", "Kitchen", "Breakfast"}`
*  `accommodates` indica el número de personas máximo que pueden ser alojadas en la propiedad. El valor es un número entero positivo de tipo `int64`
* `bathrooms` indica el número de baños que se disponen en la propiedad, el dato esta representado con `float64`.
*  `bed_type`, especifica el tipo de cama. Por ejemplo: `Real bed`, `Futon`, `Sofa`, `Airbed`, `Pull-out Sofa`, `Couch`. Este dato esta representado mediante una cadena de texto.
* `cancellation_policy`, es un valor categórico que indica la politica de cancelación: `strict`, `flexible`, `moderate`, `super_strict_30` `super_strict_60`, esta representado por una cadena de caracteres.
* `cleaning_fee`, indica si existe tarifa de limpieza. Esta representado por un tipo booleano: `True` o `False`.
* `city`, es un valor categórico ordinal que indica la ciudad donde se encuentra la propiedad. El valor esta representado por una cadena de caracteres.
* `description`, es un texto que describe la propiedad y esta representado por una cadena de caracteres.
* `first_review`, indica la fecha de la primera reseña realizada por un huésped. Es una cadena de caracteres que tiene la forma `AAAA-MM-DD`.
* `host_has_profile_pic`, esta variable indica si el host a subido una foto de perfil a la plataforma de Airbnb. El dato esta representado por una cadena de caracteres, que solo contien una `t` o `f`.
* `host_identity_verified`, este columa indica si el host a verificado su identidad en la plataforma de Airbnb. El dato esta representado por una cadena de caracteres, que solo contien una `t` o `f`.
* `host_response_rate`, mide la frecuencia con que el host respondes las consultas y las solicitudes que recibe. Esta representado en una cadena de caracteres, por ejemplo: `"100%"`, `"99%"`, `"90%"`.
* `host_since`, indica le fecha desde cuando el usuario esta registrado como host en la propiedad.  Es una cadena de caracteres que tiene la forma `AAAA-MM-DD`.
* `instant_bookable`, indica si la propiedad puede ser reservada de manera inmediata sin tener que enviar una solicitud al anfitrión y esperar a que la apruebe.  El dato esta representado por una cadena de caracteres, que solo contien una `t` o `f`.
* `last_review`, específica le fecha de la última vez que la propiedad tuvo una reseña.  Es una cadena de caracteres que tiene la forma `AAAA-MM-DD`.  
* Las variables `latitude` y `longitude`, determinan la posiciòn geospacial de la propiedad, esta con cierto rango de desviación por fines de propiedad. Ambas columnas estan representadas con un dato de tipo `float64`.
* `name`, describe el título que identifica a la propiedad en la plataforma Airbnb, tiene un formato de cadena de caracteres.
* `neighbourhood`, indica el vecindario (colonia) de la propiedad. Esta en formato de cadena de caracteres. 
* `number_of_reviews`, indica el número de reseñas que ha tenido la propiedad. El valor es un número entero positivo de tipo `int64`.
* `review_scores_rating`, especifica la calificación de las reseñas dada en una escala de 0 a 100. Esta variable esta representada con un valor de  tipo `float64`.
* `thumbnail_url`, Es una dirección url del thumbnail de la propiedad. Esta url esta representada por una cadena de caracteres.
* `zipcode`, el código postal de la dirección de la propiedad. Es un valor entero de hasta cinco dígitos que esta representado por un valor de tipo cadena de caracteres.
* `bedrooms`, indica el número de habitaciones de la propieda y esta representado por una numero de tipo `float64`.
* `beds`,  esta variable indica el número de camas en la propieday esta representado por una numero de tipo `float64`.
* `price`, especifica el costo de renta de la propiedad. Es una variable que esta representada por un dato de tipo `float64`.

En algunos casos se observa que el tipo de dato no es el correcto, ya que no es adecuado al valor que representa. Por lo que será necesario cambiar el tipo de de dato adecuado, en pasos posteriores.

* Las variabes `host_has_profile_pic`, `host_identity_verified` e `instant_bookable`, serán convertidas a tipo Booleano, ya que actualmente esta representadas por una cadena de caracteres, que solo contien una t o f.

* Las `host_response_rate`, `zipcode` estan representadas con cadena de caracteres. Por lo que se modificará a un tipo `int64`.


Para continuar con la exploración de los datos, utilzamos la indexación `loc` y `iloc`.

Por ejemplo, acomtinuación obtenemos todas las columnas del indice `179` con la ayuda de `loc`: 

In [ ]:
data.loc[179, :]

id                                                                    41118
property_type                                                         House
room_type                                                      Private room
amenities                 {TV,Internet,"Wireless Internet","Air conditio...
accommodates                                                              2
bathrooms                                                               1.0
bed_type                                                           Real Bed
cancellation_policy                                                  strict
cleaning_fee                                                           True
city                                                                     LA
description               新主人,新装修屋,干净舒适,使用中央冷暖气,非常安静,靠近公共交通,中西餐馆,超市,银行,大...
first_review                                                     2017-04-02
host_has_profile_pic                                                      t
host_identit

De manera analoga se puede realizar con `iloc`:

In [ ]:
data.iloc[179, :]

id                                                                    41118
property_type                                                         House
room_type                                                      Private room
amenities                 {TV,Internet,"Wireless Internet","Air conditio...
accommodates                                                              2
bathrooms                                                               1.0
bed_type                                                           Real Bed
cancellation_policy                                                  strict
cleaning_fee                                                           True
city                                                                     LA
description               新主人,新装修屋,干净舒适,使用中央冷暖气,非常安静,靠近公共交通,中西餐馆,超市,银行,大...
first_review                                                     2017-04-02
host_has_profile_pic                                                      t
host_identit

Se puede obtener una columna en particular con `loc` y con `iloc`:

In [ ]:
data.loc[: , 'host_response_rate']

0        100%
1         NaN
2         80%
3        100%
4        100%
         ... 
59283    100%
59284    100%
59285     33%
59286     86%
59287     84%
Name: host_response_rate, Length: 59288, dtype: object

In [ ]:
data.iloc[: , 14]

0        100%
1         NaN
2         80%
3        100%
4        100%
         ... 
59283    100%
59284    100%
59285     33%
59286     86%
59287     84%
Name: host_response_rate, Length: 59288, dtype: object

Después de seleccionar algunas filas y columnas al azar  nos encontramos con que hay datos faltantes en ciertos registros. Esto se pude confirmar desde que se ejecuto el método `info()` y la columna Non-null muestra en algunas variables muestran un valor distinto al número de observaciones (59,288).
En el caso de la variable `host_response_rate` vemos que el símbolo de `%` representa un riesgo al momento de desear realizar una conversión a tipo entero, por lo que hay que eliminar ese símbolo pare evitar errores al moemto de aplicar `casting`. Todas las columnas que representa fechas, será necesario realizar el cambio a un tipo `Date`. 

En conclusión, después del análisis de los datos se ha determinado que el conjunto de datos recopilado puede responder todas las preguntas planteadas al inicio de este trabajo. 

# Poswork 5

Después de realizar la primera exploración, es necesario realizar la limpieza del DataFrame.
Como se mencionó anteriormente algunas de las columnas en el Dataframe contienen `Nan`.
En esta sección decidiremos si aquellos `Nan` implicar eliminar una columna o solo elimiar algunos registros. Incluso, se pudee considerar la posibilidaded de llenar estos valores `NaN` con alguna valor en partícular. Verificaremos nuevamente  con la ayuda de `isna()` y la función de agregación `sum()` los datos `NaN`.

In [ ]:
data.isna().sum()

id                            0
property_type                 0
room_type                     0
amenities                     0
accommodates                  0
bathrooms                   155
bed_type                      0
cancellation_policy           0
cleaning_fee                  0
city                          0
description                   0
first_review              12662
host_has_profile_pic        151
host_identity_verified      151
host_response_rate        14558
host_since                  151
instant_bookable              0
last_review               12633
latitude                      0
longitude                     0
name                          0
neighbourhood              5513
number_of_reviews             0
review_scores_rating      13341
thumbnail_url              6520
zipcode                     789
bedrooms                     71
beds                        106
price                         0
dtype: int64

Deforma similar se puede llegar un resume para conocer el número de `NaN`en cada fila.

In [ ]:
data.isna().sum(axis=1)

0        0
1        2
2        0
3        0
4        0
        ..
59283    0
59284    0
59285    3
59286    0
59287    2
Length: 59288, dtype: int64

Comenzaremos revisando la columna `review_scores_rationg` que tiene un total de 13,341 valores `NaN` para determinar si es conveniente o no eliminarla, se debe analizar si esta columna es un determinante para el problema abordado en este tranbajo, es decir, predecir el costo que debe tener un propiedad a ser rentada en Airbnb.

45947

De aquí en adelante es lo que tenia Francisco

buuscando datos mas actuales por Last_review

In [ ]:
data.sort_values(by=['last_review'],ascending=False)

id property_type        room_type  \
6214   11164         House     Private room   
32044  54119   Condominium     Private room   
4550   33581   Condominium     Private room   
21118  35946         House     Private room   
46802  64110     Apartment     Private room   
...      ...           ...              ...   
59272   7102     Apartment     Private room   
59274  26020         House      Shared room   
59279  10637         House  Entire home/apt   
59281  14939     Apartment     Private room   
59285  14978     Apartment  Entire home/apt   

                                               amenities  accommodates  \
6214   {TV,"Wireless Internet",Kitchen,"Free parking ...             2   
32044  {"Wireless Internet","Air conditioning","Free ...             2   
4550   {"Wireless Internet","Air conditioning",Heatin...             2   
21118  {TV,"Cable TV",Internet,"Wireless Internet","A...             2   
46802  {TV,"Wireless Internet",Kitchen,"Free parking ...             2   
...                                                  ...           ...   
59272  {Internet,"Wireless Internet","Air conditionin...             2   
59274  {"translation missing: en.hosting_amenity_49",...             1   
59279  {TV,"Wireless Internet","Air conditioning",Kit...             4   
59281  {"Wireless Internet",Kitchen,"Pets live on thi...             2   
59285  {TV,Internet,"Wireless Internet","Air conditio...             3   

       bathrooms       bed_type cancellation_policy  cleaning_fee    city  \
6214         2.0  Pull-out Sofa            flexible         False  Boston   
32044        1.0       Real Bed            moderate         False  Boston   
4550         1.0       Real Bed            flexible          True  Boston   
21118        1.5       Real Bed            flexible          True  Boston   
46802        1.0       Real Bed            moderate          True  Boston   
...          ...            ...                 ...           ...     ...   
59272        1.0       Real Bed            flexible         False     NYC   
59274        1.0       Real Bed            flexible         False      LA   
59279        1.0       Real Bed              strict         False      LA   
59281        2.0       Real Bed              strict         False     NYC   
59285        1.0       Real Bed            moderate          True      LA   

       ...   longitude                                               name  \
6214   ...  -71.103835                                Cozy Private Studio   
32044  ...  -71.114579     Rm#2: 5min walk to T-metro to Boston\Cambridge   
4550   ...  -71.168449                                    The Sunset Room   
21118  ...  -71.035787                                   East Boston Room   
46802  ...  -71.134044                 Sunny, Private Rm for Female Guest   
...    ...         ...                                                ...   
59272  ...  -73.948574                 Private Room 15 min from Manhattan   
59274  ... -117.791287                            Cozy couch in San Dimas   
59279  ... -118.210777  PRODUCTION/EVENT LOCATION:Pristine Mid Century...   
59281  ...  -73.952515                     Cozy Room Upper West Manhattan   
59285  ... -118.444475                      Artsy and Spacious Back House   

            neighbourhood number_of_reviews review_scores_rating  \
6214            Hyde Park                37                 97.0   
32044       Jamaica Plain                11                100.0   
4550     Allston-Brighton                10                 98.0   
21118         East Boston                71                 97.0   
46802    Allston-Brighton                50                 98.0   
...                   ...               ...                  ...   
59272  Bedford-Stuyvesant                 0                  NaN   
59274                 NaN                 0                  NaN   
59279    Mount Washington                 0                  NaN   
59281              Harlem 

- Crear nuevas columnas en un `DataFrame`
- Reasignar columnas
- Eliminar columnas

In [ ]:
#data_2 = data.drop(columns=['BuildingArea', 'YearBuilt'])

In [ ]:
#data_2['Regionname'] = data_2['Regionname'].fillna('Unknown')

In [ ]:
#data_dropped = data_2.dropna(axis=0, how='any')

In [ ]:
data_dropped=data

Ahora, tenemos dos situaciones:

1. La primera es que nuestro índice no coincide con el número de filas que tenemos. En este caso, dado que nuestro índice es secuencial y numérico, y no tiene ningún significado además de eso, nos convendría que reflejara la cantidad de filas que tenemos en nuestro dataset.

Para lograr eso vamos a usar el método `reset_index`:

In [ ]:
data.reset_index(inplace=True)

In [ ]:
data.head(5)

index     id property_type        room_type  \
0      0  46459         House     Private room   
1      1  16468     Apartment  Entire home/apt   
2      2  30190     Apartment  Entire home/apt   
3      3  43181     Townhouse     Private room   
4      4  62109   Condominium  Entire home/apt   

                                           amenities  accommodates  bathrooms  \
0  {"Wireless Internet","Air conditioning",Kitche...             2        0.0   
1  {Internet,"Wireless Internet",Kitchen,Heating,...             2        1.0   
2  {TV,Internet,"Wireless Internet","Air conditio...             4        1.0   
3  {"Wireless Internet","Air conditioning","Pets ...             3        1.0   
4  {TV,"Cable TV",Internet,"Wireless Internet","A...             5        2.0   

   bed_type cancellation_policy  cleaning_fee  ...   longitude  \
0  Real Bed              strict          True  ...  -74.003574   
1  Real Bed              strict          True  ... -118.474525   
2  Real Bed              strict          True  ...  -73.993318   
3  Real Bed              strict          True  ... -118.096453   
4  Real Bed              strict          True  ...  -87.606271   

                                                name neighbourhood  \
0                 AFFORDABLE & COZY IN SUNSET PARK T   Sunset Park   
1  Luxurious and quiet - 30 second walk to the be...        Venice   
2                         Bensonhurst STYLISH STUDIO   Bensonhurst   
3  LA San Gabriel  Suit room 圣盖博大套房 免费停车 Free Par...   San Gabriel   
4  XTRA Large 2bdrm 2bath in the City near "El" T...      Woodlawn   

  number_of_reviews review_scores_rating  \
0                15                 89.0   
1                12                 98.0   
2                26                 94.0   
3                33                 95.0   
4                60                 88.0   

                                       thumbnail_url zipcode bedrooms beds  \
0  https://a0.muscache.com/im/pictures/105652616/...   11232      1.0  1.0   
1                                                NaN   90291      1.0  1.0   
2  https://a0.muscache.com/im/pictures/bf8ee448-b...   11214      0.0  2.0   
3  https://a0.muscache.com/im/pictures/0cff238c-a...   91776      1.0  2.0   
4  https://a0.muscache.com/im/pictures/92108903/0...   60637      2.0  2.0   

   price  
0   50.0  
1  199.0  
2   85.0  
3   70.0  
4   72.0  

[5 rows x 30 columns]

Nuestro índice ya está correcto, pero ahora tenemos un columna llamada `index` que contiene el índice original. Como no queremos guardar esos datos, agregamos la opción `drop=True` para eliminar el índice anterior:

In [ ]:
#df_dropped.reset_index(drop=True)

In [ ]:
#df_dropped = df_dropped.reset_index(drop=True)